pkg/controllers/deprovisioning/controller.go

```diff
@@ -262,12 +262,12 @@ func (c *Controller) waitForReadiness(ctx context.Context, action Command, name
 			return fmt.Errorf("getting machine, %w", err)
 		}
 		once.Do(func() {
-			c.recorder.Publish(deprovisioningevents.Launching(node, action.String()))
+			c.recorder.Publish(deprovisioningevents.Launching(machine, action.String()))
 		})
-		if _, ok := node.Labels[v1alpha5.LabelNodeInitialized]; !ok {
+		if !machine.StatusConditions().GetCondition(v1alpha5.MachineInitialized).IsTrue() {
```

Comment:
nit: can we do IsFalse() here? or does !IsTrue() also capture a nonexistent condition too?

Reply:
It captures a non-existent condition and an Unknown condition. This check shouldn't validate whether the condition is false, it should validate if the condition isn't what we need to exit. In this case, that the condition is true.